libs/model/src/aggregates/token/GetTokenInfoAlchemy.query.ts

```diff
@@ -0,0 +1,75 @@
+import { logger, type Query } from '@hicommonwealth/core';
+import { ValidChains } from '@hicommonwealth/evm-protocols';
+import { config } from '@hicommonwealth/model';
+import * as schemas from '@hicommonwealth/schemas';
+import { Network } from 'alchemy-sdk';
+import z from 'zod';
+
+const ethChainIdToAlchemy: Record<number, Network> = {
+  [ValidChains.Base]: Network.BASE_MAINNET,
+};
+
+const errorObject = {
+  network: '',
+  address: '',
+  currency: '',
+  data: [],
+};
+
+const log = logger(import.meta);
+
+export function GetTokenInfoAlchemy(): Query<
+  typeof schemas.GetTokenInfoAlchemy
+> {
+  return {
+    ...schemas.GetTokenInfoAlchemy,
+    auth: [],
+    body: async ({ payload }) => {
+      const { eth_chain_id, token_address } = payload;
+
+      const network = ethChainIdToAlchemy[eth_chain_id];
```

Comment:
The `network_id` is stored in `ChainNodes.alchemy_metadata.network_id`. Rather than hardcoding a mapping here I would prefer if we use that. It also means this route can support additional networks without needing to make code changes (just update the metadata in the DB).